### PR TITLE
profileForm UI (frontend) + connecting to backend (integeration)

### PR DESCRIPTION
**Pull request by rishabdhamija**
### What this PR does (required):
- This PR involves creating a Ui for profile Form as well as storing submitted data to the backend in profiles collection.
- The profile collection is updated based on the latest instructions about combining dogOwner as well as dogSitter into the same collection. (inserted UserID field into profile collection).
- A few fields relevant to dogSitter are made non-required in profile collection as well as in formik validation frontend, since those fields would be empty in case user chooses to be dogOwner and not dogSitter.

### Screenshots / Videos (required):
- Initially the user is presented with a choice whether he is a animalSitter. If no, then the following form is presented.
Screen Shot 2021-06-23 at 10 36 45 AM

- If user chooses yes as dogSitter, then additional formFields only relevant to dogSitter are presented as shown below:
Screen Shot 2021-06-23 at 10 37 09 AM

- MADE CHANGES IN PROFILE MODEL ACCORDINGLY. Inserted UserID field referencing user model. Before save, the pre(save) method is invoked. It searches user collection for emailID (assuming email will be unique), grabs the userID and inserts it into profile.userID position (for referencing later).
Screen Shot 2021-06-23 at 1 05 46 PM
 
- this is how the database will now save a profile.
Screen Shot 2021-06-23 at 1 09 33 PM

### Any information needed to test this feature (required):
- localhost:3000/dogSitter is the endpoint for testing this feature. 

### Any issues with the current functionality (optional):
- Note: this feature works without authentication in frontend (it will be added at later point)

### CODE REVIEW

File: client/src/helpers/APICalls/createProfile.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      lastName,
+      email,
+      phoneNumber,
+      serviceCharge,
+      address,
+      description,
+    }),
+    credentials: 'include',
+  };
+  // console.log('This is the fetchOptions body', fetchOptions.body);
```
Comment (sina-jamshidi) [inline client/src/helpers/APICalls/createProfile.ts:31]: remove all commented code and any console.logs

File: server/routes/auth.js [hunk 2]
```diff
@@ -17,4 +18,8 @@ router.route('/user').get(protect, loadUser);
 
 router.route('/logout').get(logoutUser);
 
+
+// sample register routes
+router.route('/createProfile').post(createProfile);
```
Comment (sina-jamshidi) [inline server/routes/auth.js:23]: it probably makes sense to create a profile in register itself. We shouldn't need a separate route

**Review by sina-jamshidi — APPROVED** [review]
I've already seen the code from when we worked on it. Should be good to go with a couple of small comments

File: server/controllers/auth.js [hunk 3]
```diff
@@ -114,3 +115,43 @@ exports.logoutUser = asyncHandler(async (req, res, next) => {
 
   res.send("You have successfully logged out");
 });
+
+
+// @route POST /auth/profile
```
Comment (sina-jamshidi) [inline server/controllers/auth.js:120]: the route doesn't make much sense to be `auth/profile`
profile doesn't have anything to do with auth

File: client/src/helpers/APICalls/createProfile.ts (hunk 1 above)
Comment (rishabdhamija) [reply client/src/helpers/APICalls/createProfile.ts:31]: removed comments as well as ocnsole.log() statements used for testing

**Comment by rishabdhamija** [comment]
merging profileForm FE + Integeration

**Pull request merged**